crates/api/src/config.rs

```diff
@@ -14,68 +15,236 @@ fn tc<S: serde::Serialize, D: serde::de::DeserializeOwned>(
     .map_err(|e| K2Error::other_src("decode", e))
 }
 
-/// Denotes a type used to configure a specific kitsune2 module.
-///
-/// Note, the types defined in this struct are specifically for configuration
-/// that cannot be changed at runtime, the likes of which might be found
-/// in a configuration file.
-///
-/// If a specific module has a config that can be changed at runtime, the
-/// component found in this type might be a `default_` prefixed version
-/// of it, then the runtime value can be altered through different means.
-///
-/// It is highly recommended that you expose this struct in your module
-/// docs to help devs using your module understand how to configure it.
-pub trait ModConfig:
-    'static
-    + Sized
-    + Default
-    + std::fmt::Debug
-    + serde::Serialize
-    + serde::de::DeserializeOwned
-    + Send
-    + Sync
-{
+/// A callback to be invoked if the config value is updated at runtime.
+pub type ConfigUpdateCb =
+    Arc<dyn Fn(serde_json::Value) + 'static + Send + Sync>;
+
+#[derive(Clone, serde::Serialize, serde::Deserialize)]
+#[serde(transparent, rename_all = "camelCase")]
+struct ConfigEntry {
+    pub value: serde_json::Value,
+    #[serde(skip, default)]
+    pub update_cb: Option<ConfigUpdateCb>,
+}
+
+impl std::fmt::Debug for ConfigEntry {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.value.fmt(f)
```

Comment:
Might be nice to include the presence or absence of the `update_cb` even though it's not meaningful to log the value?

Reply:
Done: https://github.com/holochain/kitsune2/pull/58/commits/eb1137054b4208e7ba66e65f3d20bd4e8acd9c0a